codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -236,7 +236,8 @@ private void writePendingBytes() {
 
     /**
      * This will allocate bytes by stream weight and priority for the entire tree rooted at {@code parent}, but does
-     * not write any bytes.
+     * not write any bytes. Priority is generally distributed proportionally to siblings in the priority tree, however
+     * we ensure that at least 1 byte is assigned to each stream in order to make progress.
```

Comment:
@nmittler 

> we ensure that at least 1 byte is assigned to each stream in order to make progress.

This may be a bit misleading.  We should be ensuring that we use the entire connection window is used (assuming streams have `>= connection window bytes` to send) and we may need some sort of rounding to accomplish this.  I think we are attempting to use the `Math.max(1, ..)` mechanism to accomplish this but I'm not sure this will do the trick.  For example if we have 2 streams that end up getting `22.2` and `22.8` due to their respective weights...I think with this code we will allocate `22` and `22` respectively. Do we have an extra mechanism to ensure that extra `1` will be assigned to some stream?